doc/source/data/dataset.rst

```diff
@@ -327,6 +327,134 @@ By default, transformations are executed using Ray tasks. For transformations th
     # Save the results.
     ds.repartition(1).write_json("s3://bucket/inference-results")
 
+Last-mile preprocessing
+-------------------------------------
+
+Datasets supports data preprocessing transformations commonly performed just before model training and model inference, which we refer to as **last-mile preprocessing**. These transformations are carried out via a few key operations: mapping, groupbys + aggregations, and random shuffling.
+
+Mapping
+~~~~~~~
+
+Many common preprocessing transformations, such as:
+- adding new columns
+- transforming existing columns
+- dropping columns
+- dropping nulls
+- one-hot encoding
+
+can be efficiently applied to a dataset using Pandas DataFrame UDFs and ``.map_batches()``; this will do these transformations in parallel over the dataset's blocks, and will allow you to apply vectorized Pandas operations to columns.
+
+.. code-block:: python
+
+    def transform_batch(df: pd.DataFrame):
+        # Drop nulls.
+        df = df.dropna(subset=["feature_1"])
+        # Add new column.
+        df["new_col"] = df["feature_1"] - 2 * df["feature_2"] + df["feature_3"] / 3
+        # Transform existing column.
+        df["feature_1"] = 2 * df["feature_1"] + 1
+        # Drop column.
+        df.drop(columns="feature_2", inplace=True)
+        # One-hot encoding.
+        fruits = ["apple", "orange", "banana"]
+        for fruit in fruits:
+            df[f"fruit_{fruit}"] = df[fruit].map(
+                collections.defaultdict(int, **{fruit: 1}))
```

Comment:
The columns in this transformation are disjoint from the columns in the groupby + aggregations and random shuffling sections, could unify these to a common data model if that would be cleaner for the user to digest.

Reply:
could we make the column something like "state_<state abbr>" or something? might be clearer to user. fruit just comes out of nowhere

Reply:
I think that I'll change this to `categories = ["cat_1", "cat_2", "cat_3"]` and `df[f"categorical_{cat}"]`, to make it extra clear that this is a categorical column.